x/twap/strategy_test.go

```diff
@@ -211,28 +235,69 @@ func (s *TestSuite) TestComputeGeometricStrategyTwap() {
 		"accumulator = 10*OneSec, t=100s. .1*second base accum": geometricTestCaseFromDeltas0(
 			s, OneSec.MulInt64(10).Mul(logOneOverTen), geometricTenSecAccum, 100*time.Second, twap.TwapPow(geometricTenSecAccum.QuoInt64(100*1000))),
 
-		// TODO: this is the highest price we currently support with the given precision bounds.
-		// Need to choose better base and potentially improve math functions to mitigate.
 		"price of 1_000_000 for an hour": {
 			startRecord: newOneSidedGeometricRecord(baseTime, sdk.ZeroDec()),
 			endRecord:   newOneSidedGeometricRecord(baseTime.Add(time.Hour), OneSec.MulInt64(60*60).Mul(twap.TwapLog(sdk.NewDec(1_000_000)))),
 			quoteAsset:  denom0,
 			expTwap:     sdk.NewDec(1_000_000),
 		},
-		// TODO: overflow tests
-		// - max spot price
-		// - large time delta
-		// - both
 
-		// TODO: hand calculated tests
+		"no overflow test: at max spot price denom0 quote - get max spot price": {
+			startRecord: withSp0(baseRecord, gammtypes.MaxSpotPrice),
+			endRecord:   newOneSidedGeometricRecord(baseRecord.Time.Add(oneHundredYearsMin1MsDuration), overflowTestCaseAccumDiff),
+			quoteAsset:  denom0,
+			expTwap:     gammtypes.MaxSpotPrice,
+		},
+
+		"expected precision loss test: - at spot price denom1 quote - panic": {
+			startRecord: withSp0(baseRecord, gammtypes.MaxSpotPrice),
+			endRecord:   newOneSidedGeometricRecord(baseRecord.Time.Add(oneHundredYearsMin1MsDuration), overflowTestCaseAccumDiff),
+			quoteAsset:  denom1,
+
+			// TODO: analyze if panic is acceptable.
+			// Should be acceptable since underlying logic is
+			// only called from the query path.
+			expPanic: true,
+		},
```

Comment:
I don't understand why this is panicking?

Reply:
Ah we discussed, its due to divide by 0 right here: https://github.com/osmosis-labs/osmosis/blob/5d360857e24cf1aa7c0d465fd9d16df6bb202357/x/twap/strategy.go#L53

Discussed to just add an if statement there

Reply:
Upon looking into this more, I think this might have been a panic caused by the test setup.

I updated the assertion of this test case to return zero, as we discussed, and it passes.

Given the current test cases, I don't think it is possible for that value to be zero. This gets executed only if the arithmetic mean of spot prices is negative which happens when spot prices are in range (0, 1). 

I will still add this check as a defense-in-depth but I can't think of a test case that would cover this logic

Reply:
Also, FYI that I'm working on reducing the number of these multiplicative inverses that can cause the precision loss here:
https://github.com/osmosis-labs/osmosis/pull/3924